update.py

```diff
@@ -88,9 +89,16 @@ def handle_notices_diff(notices):
     print 'Checking ', len(notices), 'notices'
     for notice in notices:
         sanitised_notice = sanitise_notice_for_database(notice)
-        db_notice = notices_coll.find_one(sanitised_notice)
+        notice_cpy = shallow_copy(sanitised_notice)
+        try:
+            del notice_cpy['uid']
+        except KeyError:
+            pass
+
+        db_notice = notices_coll.find_one({'$or':[{'uid' : sanitised_notice['uid']}, notice_cpy]})
         if db_notice is None:
-            different_notices.append(notice)
+            different_notices.append(sanitised_notice)
```

Comment:
Hey, @iakshat Can you please help me understand why we have changed from `notice` to `sanitised_notice`?

Reply:
I changed it just to make the logs readable if we require some time in the future. Otherwise, it prints a very very long raw file that I don't think is required there.

Reply:
Oh, that is a keen observation. Thanks :D

Reply:
However, do you think this will cause issues and not send properly the data? If you look at like 106, we are already sanitizing the notice before adding it to the database. And line 107, I believe the `hooks.notices_updated` function (I'm not sure though as haven't gone through the code) expects an unsanitised notice. So do you think it's best not to change it for now?

Reply:
ahh yes! I should have checked it properly😖. I'll change the code to include both log and hooks factors.